ironfish/src/node.ts

```diff
@@ -226,13 +234,35 @@ export class IronfishNode {
     }
     const workerPool = new WorkerPool({ metrics, numWorkers: workers })
 
-    const consensus = new TestnetParameters()
+    metrics = metrics || new MetricsMonitor({ logger })
+
+    let networkDefinitionJSON = ''
+    // Try fetching custom network definition first, if it exists
+    if (config.get('customNetwork') !== '') {
+      networkDefinitionJSON = await files.readFile(files.resolve(config.get('customNetwork')))
+    } else if (config.get('networkId') === 0) {
+      networkDefinitionJSON = TESTING
+    } else if (config.get('networkId') === 1) {
+      networkDefinitionJSON = TESTNET_PHASE_2
+    }
+
+    const networkDefinition = IJSON.parse(networkDefinitionJSON) as {
+      id: number
+      bootstrapNodes: string[]
+      genesis: SerializedBlock
+      consensus: ConsensusParameters
+    }
```

Comment:
It'd be nice to have a `yup` schema for this to validate the incoming type like we do for the config: https://github.com/iron-fish/ironfish/blob/22e9f8ce5cded0ce32ecdce8750869111638cb23/ironfish/src/fileStores/config.ts#L250

That way we get an error up front if the input is incorrectly formatted rather than somewhere else in the node. I realize this might be a bit inconvenient to do since there are several fields in the genesis block data + consensus params, but still think it'd be worthwhile.

Reply:
Added `yup` validation